Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -721,8 +721,13 @@ dtStatus dtNavMeshQuery::findNearestPoly(const float* center, const float* exten
 	if (dtStatusFailed(queryPolygons(center, extents, filter, polys, &polyCount, MAX_SEARCH)))
 		return DT_FAILURE | DT_INVALID_PARAM;
 	
+	if (polyCount == 0)
+		return DT_SUCCESS;
```

Comment:
That's still the behavior - `nearestRef` is assigned to 0 in the beginning of the function.